File: InterviewPratice/Java/IsCryptSolution.java
final class IsCryptSolution {

    boolean isCryptSolution(String[] crypt, char[][] solution) {
        for(int i = 0; i < 3; i++)
            for(char[] c : solution) crypt[i] = crypt[i].replaceAll(c[0]+"", c[1]+"");

        for(String s : crypt) if(s.matches("0+\\d+")) return false;
        return Double.parseDouble(crypt[0]) + Double.parseDouble(crypt[1])
        == Double.parseDouble(crypt[2]);
    }
}

File: InterviewPratice/Javascript/isCryptSolution.js
function isCryptSolution(crypt, solution) {
    for(let i = 0; i < 3; i++) {
        for(let c of solution)
            crypt[i] = crypt[i].replace(new RegExp(c[0], "g"), c[1]);
    }
    
    for(let s of crypt) if(/^0+\d+/g.test(s)) return false;
    
    return parseInt(crypt[0]) + parseInt(crypt[1]) == parseInt(crypt[2]);
}